File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s = strs[0];
        
        int global_min = INT_MAX;
        
        for (size_t i = 1; i < strs.size(); i++) {
            string x = strs[i];
            int current_max = 0;
            // cout << x << " ";
            for (int j = 0; j < min(s.size(), x.size()); j++) {
                if (x[j] == s[j]) current_max++;
                else break;
            }
            // cout << i << " " << current_max << endl;
            global_min = min(global_min, current_max);
        }
        
        return s.substr(0, global_min);
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        i = 0
        
        for chars in zip(*strs):
            
            if len(set(chars)) == 1:
                i += 1
            else:
                break
        
        return strs[0][:i]
        